src/JsonSchema/Constraints/BaseConstraint.php

```diff
@@ -89,4 +90,22 @@ public function reset()
     {
         $this->errors = array();
     }
+
+    /**
+     * Recursively cast an associative array to an object
+     *
+     * @param array $array
+     *
+     * @return object
+     */
+    public static function arrayToObjectRecursive($array)
+    {
```

Comment:
I think it is worth considering the performance in this case. Perhaps a continuous means of measuring performance can be added in another PR to facilitate a continuous improvement and scrutiny?